### PR TITLE
resolve: issues (fix #434, fix #421, fix #437)

### LINKED ISSUE #421: popupSave faile if I change other color calendar
## Version
v1.12.5

## Development Environment
Google Chrome 
 77.0.3865.90

## Current Behavior
I used the [online demo]( https://nhn.github.io/tui.calendar/latest/tutorial-example00-basic)
I can modify the event's subtitle, content, start and end time, but nothing will change after clicked the UPDATE button, only if I change the calendar dropdown list at the same time.
## Expected Behavior
The calender of event could be update.

### LINKED ISSUE #434: changing schedule.category from 'time' to 'allday' doesn't move to all day row, even after updateSchedule()
Hi,

When changing schedule.category from 'time' to 'allday' and invoking updateSchedule, the schedule is NOT moved to all day row (other updates work). If reloading the page, then schedule shows properly.

### LINKED ISSUE #437: bug + fix: calendar colors always overwrite schedule colors
Need: we need some schedules to be in custom colors, different than the calendar they belong to.
This should work according to docs, since each schedule can have it's own colors.

Bug: calendar colors always overwrite schedule colors, instead of letting those alone if they were set in schedule data.

Fix: change the code block at
https://github.com/nhn/tui.calendar/blob/188b1846eb1023660e0776ef0fcfc1a4846e9b64/src/js/factory/calendar.js#L757

to
```
        //overwrite only if not customized
        if (color) {
            if (typeof obj.color == 'undefined') { obj.color = color.color; }
            if (typeof obj.bgColor == 'undefined') { obj.bgColor = color.bgColor; }
            if (typeof obj.borderColor == 'undefined') { obj.borderColor = color.borderColor; }
        }
```

I tested that this fix works for me.

Please make to commit this in v2 too.

Thanks

### LINKED ISSUE #452: 24 hour time format required for WEEK and DAY views.
## Version
latest

## Development Environment
Archlinux
webpack
vuejs
npm
code
firefox

## Current Behavior
Calendar week view is showing time in 12 hour format (1-12 AM/PM).

## Expected Behavior
I would expect to change some setting to select whether I want to show hours in week view in 24 or 12 hour format. It's the same in day view too.
In my case the calendar is not meant for US users, but rather some European country, so time format required is 24 hour.
(I noticed that current time (marked with purple line) is showing 24hour time.)

### PR DESCRIPTION
**Pull request by jungeun-cho**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)
- [x] It does not introduce a breaking change or has description for the breaking change

### Description
* fix #434 : When creating a schedule, passing the category property to 'allday' does not create an all-day schedule.
* fix #421 : When changing the calendar ID in the default edit popup, the calendar is not reflected.
* fix #437 : If a calendar color exists, overwriting the calendar color even if you pass the custom color of the schedule as a property.
* fix #452 : fix: resolve handlebar template fuction override
    * It's an issue with the [react wrapper](https://github.com/nhn/toast-ui.react-calendar/issues/18).

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

**Comment by mariusa** [comment]
Suggestion: ```isAllDay``` should't be a settable property of schedule.
If needed for convenience, it should be a computed property (in Vue, not sure what's the JS equivalent)
```
isAllDay = (options.category && options.category === 'allday')
```

File: src/js/view/popup/scheduleCreationPopup.js [hunk 1]
```diff
@@ -295,7 +295,8 @@ ScheduleCreationPopup.prototype._onClickSaveSchedule = function(target) {
                 isAllDay: isAllDay,
                 state: state.innerText,
                 triggerEventName: 'click',
-                id: this._schedule.id
+                id: this._schedule.id,
+                _feId: this._schedule.__fe_id
```
Comment (dongsik-yoo) [inline src/js/view/popup/scheduleCreationPopup.js]: Schedule.prototype.cid / ScheduleViewModel.prototype.cid

Comment (jungeun-cho) [reply src/js/view/popup/scheduleCreationPopup.js]: @dongsik-yoo 
If the user does not use the default pop-up UI, or if the user directly calls the `updateSchedule` method, `cid` does not exist.
So, to find the schedule in the schedule collection, I modified it to pass the calendar Id (`originCalendarId`) of the schedule before the change.

**Comment by jungeun-cho** [comment]
@mariusa 
Even if the category is time, if it is more than one day (24 hours), it is considered as an all-day schedule.

When specifying category as `allday`, it is better to turn on `isAllDay` option.

File: src/js/common/common.js [hunk 2]
```diff
@@ -320,5 +321,21 @@ module.exports = {
         }, contextopt);
 
         return found;
+    },
+
+    getChangesSchedule: function(schedule, propNames, data) {
+        var changes = {};
+
+        util.forEach(propNames, function(propName) {
+            if (['start', 'end'].indexOf(propName) > -1) {
```
Comment (dongsik-yoo) [inline src/js/common/common.js]: This array must be declared out of loop.

File: src/js/controller/base.js [hunk 3]
```diff
@@ -155,6 +155,18 @@ Base.prototype.updateSchedule = function(schedule, options) {
 
     options = options || {};
 
+    if (options.category && options.category === 'allday') {
```
Comment (dongsik-yoo) [inline src/js/controller/base.js]: The condition `options.category` is not necessary.

File: src/js/factory/calendar.js [hunk 4]
```diff
@@ -787,18 +780,18 @@ Calendar.prototype.getSchedule = function(scheduleId, calendarId) {
 /**
  * Update the schedule
  * @param {string} scheduleId - ID of a schedule to update
- * @param {string} calendarId - The calendarId of the schedule to update
+ * @param {string} calendarId - The calendarId of the schedule before change
  * @param {Schedule} scheduleData - The {@link Schedule} data to update
```
Comment (dongsik-yoo) [inline src/js/factory/calendar.js]: You must add `changes` property of this event.

File: src/js/handler/daygrid/move.js [hunk 5]
```diff
@@ -229,7 +234,7 @@ DayGridMove.prototype._onDragEnd = function(dragEndEventData, overrideEventName,
         dragEnd: this._onDragEnd,
         click: this._onClick
     }, this);
-
+    console.log(dragEndEventData.originEvent, scheduleData);
```
Comment (dongsik-yoo) [inline src/js/handler/daygrid/move.js]: Remove unnecessary log.

File: src/js/common/common.js [hunk 6]
```diff
@@ -320,5 +321,21 @@ module.exports = {
         }, contextopt);
 
         return found;
+    },
+
+    getChangesSchedule: function(schedule, propNames, data) {
```
Comment (dongsik-yoo) [inline src/js/common/common.js]: how about `getScheduleChanges()`?

File: src/js/handler/time/resize.js [hunk 7]
```diff
@@ -215,15 +217,22 @@ TimeResize.prototype._updateSchedule = function(scheduleData) {
...
+    );
+
     /**
      * @event TimeResize#beforeUpdateSchedule
      * @type {object}
-     * @property {Schedule} schedule - schedule instance to update
-     * @property {Date} start - start time to update
-     * @property {Date} end - end time to update
+     * @property {Schedule} schedule - The original schedule instance
+     * @property {object} changes - end time to update
```
Comment (dongsik-yoo) [inline src/js/handler/time/resize.js]: The legacy properties' JSDoc must not be removed. Mark it as deprecated. `calendar.js` file is too.

File: src/js/handler/daygrid/move.js (hunk 5 above)
Comment (jungeun-cho) [reply src/js/handler/daygrid/move.js]: Yeah.

File: src/js/handler/time/resize.js (hunk 7 above)
Comment (jungeun-cho) [reply src/js/handler/time/resize.js]: right. I'll fix it.

**Comment by jungeun-cho** [comment]
ok to test

**Comment by jungeun-cho** [comment]
ok to test

**Pull request merged**